lisa/sut_orchestrator/libvirt/schema.py

```diff
@@ -52,6 +52,26 @@ class BaseLibvirtPlatformSchema:
     capture_libvirt_debug_logs: bool = False
 
 
+# Configuration options for device-passthrough for the VM.
+@dataclass_json()
+@dataclass
+class LibvirtHostPciDeviceSchema:
+    # This property determines if libvirt would bind/unbind native driver or not
+    # Default is set to "yes" means libvirt will detach the native device driver and
+    # attach it to vfio-pci when vm is create and attach it back on deletion of vm
+    managed_passthrough: str = "yes"
+
+    # Type of the device. Default is set to "pci"
+    device_type: str = "pci"
+
+    # Host device details for which we want to perform device-passthrough
+    # we can get it using lspci command
+    source_domain: str = ""
```

Comment:
these params are also applicable to guest PCI device. Those are optional to pass in libvirt. There will be 2 `address `element for `hostdev`. One under `source `( for host device) and other is for guest vm. So keeping **source_** seems fine to me

Reply:
Could you provide a sample xml to understand more? 
Does the other guest related `address` also have same domain, bus, slot, function fields?

Reply:
Thanks, Smit, for helping me understand the schema in detail. 

If we also plan to extend `LibvirtHostPciDeviceSchema` to specify the other guest specific `address` as well, then a better schema would be to have a new class called `PCIDeviceSchema` that will have domain, bus, slot, function. And have say `source_device` and `guest_device` in `LibvirtHostPciDeviceSchema`
```
class PCIDeviceSchema:
  domain: str
  bus: str
  slot: str
  function: str

class LibvirtHostPciDeviceSchema:
  .
  .
  .
  source_device: PCIDeviceSchema
  guest_device: PCIDeviceSchema

```

Reply:
As discussed, this will make guest_device config mandatory to be passed from runbook. We can make it optional but we can extend the platform whenever that need will come (Say some post domain creation testcase which check the LSPCI output for guest to verify). Idea with this PR is to enable PCI passthrough.